pkg/query-service/app/http_handler.go

```diff
@@ -2188,6 +2189,151 @@ func (aH *APIHandler) WriteJSON(w http.ResponseWriter, r *http.Request, response
 	w.Write(resp)
 }
 
+// Preferences
+
+func (ah *APIHandler) RegisterPreferenceRoutes(router *mux.Router, am *AuthMiddleware) {
+	subRouter := router.PathPrefix("/api/v1/preferences").Subrouter()
+
+	subRouter.HandleFunc("/user", am.ViewAccess(ah.getUserPreference)).Methods(http.MethodGet)
+
+	subRouter.HandleFunc("/user", am.ViewAccess(ah.updateUserPreference)).Methods(http.MethodPost)
+
+	subRouter.HandleFunc("/user/all", am.ViewAccess(ah.getAllUserPreference)).Methods(http.MethodGet)
+
+	subRouter.HandleFunc("/org", am.AdminAccess(ah.getOrgPreference)).Methods(http.MethodGet)
+
+	subRouter.HandleFunc("/org", am.AdminAccess(ah.updateOrgPreference)).Methods(http.MethodPost)
+
+	subRouter.HandleFunc("/org/all", am.AdminAccess(ah.getAllOrgPreference)).Methods(http.MethodGet)
+
+}
+
+func (aH *APIHandler) getAllUserPreference(w http.ResponseWriter, r *http.Request) {
+
+	orgId := r.URL.Query().Get("orgId")
```

Comment:
Don't extract the organization ID (orgId) from URL parameters. Instead, retrieve it from the context. Relying on URL params requires additional verification to ensure the user actually belongs to that organization. The only IDs you should trust from the client are those they're actively working with. Otherwise, they could make speculative requests that might leak information or cause incorrect updates.
Consider a future scenario where we migrate to PostgreSQL, storing all data in a single database with multiple orgIds. In this case, relying on client-provided orgIds would be even riskier.

The only valid reason to accept an orgId from the client would be if we eventually support multiple organizations within a single entity/company, allowing admins to update across organizations. However, this is a distant and vague future possibility. Even in such a case, we'd need robust checks to verify the user's permissions for the orgIds they're attempting to modify.

Reply:
makes sense, kept it as a param to support multiple orgs case but i understand it will require extra perms check, will move to context check

Reply:
done